File: frameworks/shared/ruby/lib/velocitybench/healthcheck/health_status.rb
# frozen_string_literal: true

module VelocityBench
  module HealthCheck
    # Health check status values.
    module HealthStatus
      UP = 'up'
      DEGRADED = 'degraded'
      DOWN = 'down'
      IN_PROGRESS = 'in_progress'

      ALL = [UP, DEGRADED, DOWN, IN_PROGRESS].freeze

      def self.valid?(status)
        ALL.include?(status)
      end
    end
  end
end

File: frameworks/shared/csharp/VelocityBench.HealthCheck/HealthStatus.cs
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace VelocityBench.HealthCheck;

/// <summary>
/// Health check status values.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum HealthStatus
{
    [EnumMember(Value = "up")]
    Up,

    [EnumMember(Value = "degraded")]
    Degraded,

    [EnumMember(Value = "down")]
    Down,

    [EnumMember(Value = "in_progress")]
    InProgress
}
